src/test/resources/commonutil/InputCommonUtilTest_resource.txt

```diff
@@ -0,0 +1 @@
+good
```

Comment:
Why does this file need to be in its own folder and not part of checkstyle's standard packaging?

Reply:
Because it most elegantly illustrates the initial issue I encountered and have now fixed - which is to reach for a file outside of the package you're currently in.
I _could_ move everything inside the "standard packaging", but at the expense of adding an excessively deep directory tree.
The "good" file would be then in `src/test/resources/com/puppycrawl/tools/checkstyle/utils/commonutil`, the "bad" file would be in `src/test/resources/com/puppycrawl/tools/checkstyle/utils/commonutil/com/puppycrawl/tools/checkstyle/utils/commonutil`, which would be a lot less comprehensive and a less good representation of the limitation I'm trying to illustrate.

Reply:
@romani What do you think of this?

Reply:
> I could move everything inside the "standard packaging", but at the expense of adding an excessively deep directory tree.

please deep path. Nobody will look at it, till deal with certain test.

Reply:
This is not so easy to fix after all. A good test must first _fail_ without the fix (illustrating the issue) and _succeed_ with the fix. The easy part is to make the test succeed with the fix. The hard part is to make it fail without the fix. That's why I have the two files by the way, the "good" file and the "bad" file. Whatever I try, there is always a combination where one of these files causes the `AllTestsTest` to fail. That is the reason it's taking me so long to fix this.

Reply:
I tried, but I couldn't move the resources while still keeping `AllTestsTest` happy, and I certainly don't want to reduce the value of my test, so I left it the way it was because it's the only way I could preserve the value of the test while still satisfying `AllTestsTest`.

Reply:
@peterdemaeyer what failure were you getting in `AllTestsTest` and what path were you setting it to?
The only reason `AllTestsTest` isn't flagging this now is because it is specifically looking in `com/puppycrawl` sub-folder which your file is not in.
You can add suppression to test and we can review later.
https://github.com/checkstyle/checkstyle/blob/a73ff0890c27cbe8affc1a55f2710a231687bd85/src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java#L153
https://github.com/checkstyle/checkstyle/blob/a73ff0890c27cbe8affc1a55f2710a231687bd85/src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java#L255